crates/services/p2p/src/peer_manager.rs

```diff
@@ -553,6 +633,41 @@ impl PeerManager {
         }
     }
 
+    fn batch_update_score(&mut self, score: PeerScore) {
+        for (peer_id, peer_info) in &mut self.non_reserved_connected_peers {
+            let new_score = peer_info.score + score;
```

Comment:
Instead of `peer_info.score + score` maybe it worth to do something like `peer_info.score * DECAY_FACTOR` where `DECAY_FACTOR < 1`. In this case, it will be not possible to disconnect during decaying=)